### PR TITLE
Move `__stdin.d` special case to FileManager and give it memory ownership

### PR DESCRIPTION
**Pull request by Geod24**
A second step in getting rid of the IO in `Module`, this moves the `__stdin.d` hack inside of `FileManager` so the API is uniform to callers. By acknowledging that the ownership of the memory is in `FileManager`, we can also remove some odd special casing which, I believe, wouldn't have worked anyway (because `lookup` would have returned `null`).

### CODE REVIEW

File: src/dmd/dmodule.d [hunk 1]
```diff
@@ -658,13 +658,13 @@ extern (C++) final class Module : Package
      */
     bool read(const ref Loc loc)
     {
-        if (srcBuffer)
+        if (this.src)
             return true; // already read
```
Comment (ibuclaw) [inline src/dmd/dmodule.d:662]: So one only needs to set `src` if the reading from stdin is already taken care of?  I think this fine.

**Review by ibuclaw — APPROVED** [review]

**Comment by ibuclaw** [comment]
BTW, I don't think we need either the file_manager.h nor root/file.h headers anymore after this.

File: src/dmd/dmodule.d (hunk 1 above)
Comment (Geod24) [reply src/dmd/dmodule.d:662]: Yeah that's a good point, and would be an argument to leave it in `mars.d` 🤔 
Do you use `readFromStdin` in GDC ?

**Comment by Geod24** [comment]
Updated / fixed now that Github works again, should be ready to go. Next step will be to make `Module` contains a `const(char)[]` for `src` @ibuclaw

**Comment by ibuclaw** [comment]
Going to remove the headers here too?

**Comment by Geod24** [comment]
Well maybe someone wants to use those types from C++ ? :)
As long as we don't have a (clearly) defined C++ interface an argument can be made both ways.

**Comment by ibuclaw** [comment]
> Well maybe someone wants to use those types from C++ ? :)
> As long as we don't have a (clearly) defined C++ interface an argument can be made both ways.

There are only two downstream users, and the interface has already been stripped down to only what they need to invoke and compile code from the front-end, so what they goes.

If this is now internalised - and it looks that way - then the entire struct(s) can just be `extern (D)`

**Comment by Geod24** [comment]
Faire enough, added a commit.

**Pull request merged**